sys/net/link_layer/ieee802154/submac.c

```diff
@@ -358,13 +358,22 @@ int ieee802154_set_phy_conf(ieee802154_submac_t *submac, uint16_t channel_num,
         return -ENETDOWN;
     }
 
-    int res = ieee802154_radio_config_phy(dev, &conf);
+    int res;
+    if ((res = ieee802154_radio_request_set_trx_state(dev, IEEE802154_TRX_STATE_TRX_OFF)) < 0) {
```

Comment:
Would it be possible that this line is reached when the TRX state is already TRX_OFF? The API doc doesn't define whether `ieee802154_radio_request_set_trx_state()` should succeed or fail, if the new state is equal to the current state. We might want to add this to ensure consistent behavior (or at least know who to blame when behavior is not consistent).

Reply:
The question here is, should we return "-EALREADY" in this function? Because the radio won't malfunction if we set the same state again.
We could also specify in the API that `set_trx_state` shouldn't be called to set the same state.

What do you think?

Reply:
So, how do we proceed here?

1) Do nothing and assume it's ok to set the transceiver state even if the radio is already in that state
2) Return ' -EALREADY` if the radio is in the state that the `set_trx_state` function tries to set
3) Just add this constraint as documentation, indicating that the upper layer must be sure that the radio is not already in that state.

Which one?

Reply:
ping @maribu @benpicco @fjmolinas @PeterKietzmann

Reply:
If the driver can know that the device is already in the state (option 2) it might as well either skip the instruction if the radio would have an issue with that or simply call it again if it is no issue. (option 1) so those two could be seen as equivalent.

I think most radios would simply ignore it, so 1) seems like the easiest way to go.

Reply:
This seems reasonable

Reply:
then, this original snippet should stay as it is